packages/drivers/odsp-driver/src/odspDelayLoadedDeltaStream.ts

```diff
@@ -181,13 +181,14 @@ export class OdspDelayLoadedDeltaStream {
 						this.cache.sessionJoinCache.remove(this.joinSessionKey);
 					}
 					// If we hit this assert, it means that "disconnect" event is emitted before the connection went through
-					// dispose flow which is not correct and could lead to a bunch of erros.
+					// dispose flow which is not correct and could lead to a bunch of errors.
 					assert(connection.disposed, 0x4ae /* Connection should be disposed by now */);
 					this.currentConnection = undefined;
 				});
 				this.currentConnection = connection;
 				return connection;
 			} catch (error) {
+				this.clearJoinSessionTimer();
```

Comment:
@jatgarg wasn't able to reproduce this leak. But going through the code, there is the possibility of this block throwing and the caller (OdspDocumentService.connectToDeltaStream) to abandon the newly created object which would still have a timer scheduled. Please let me know if you think this might be it. Anyway, we would never schedule this twice for the same object and if we do, we will catch it in the future.

Reply:
I think this PR is good as a first step. It could either solve or provide more information on the matter. We do dispose the timer on service dispose.

Reply:
We do clear the timer on dispose but I'm not sure disposed() is called in the code path I mentioned.

My main concern here would be that we're sweeping under the rug some unknown weird object lifecycle but that log statement + long stack would give us more information.